components/authz-service/storage/postgres/migration/sql/54_use_project_db_id_in_references_of_project_rules.up.sql

```diff
@@ -0,0 +1,313 @@
+BEGIN;
+-- break all references
+ALTER TABLE iam_project_rules
+    DROP CONSTRAINT iam_project_rules_project_id_fkey;
+ALTER TABLE iam_staged_project_rules
+    DROP CONSTRAINT iam_staged_project_rules_project_id_fkey;
+
+-- When we're done fixing all references, we'll change the
+-- primary key of iam_projects; now, we only add a uniqueness
+-- constraint on what's going to be the new PK: db_id.
+ALTER TABLE iam_projects
+    ADD UNIQUE (db_id);
+
+-- redo references (welcome to copy-paste hell)
+ALTER TABLE iam_project_rules RENAME COLUMN project_id TO project_temp_id;
+ALTER TABLE iam_project_rules
+    ADD COLUMN project_id INTEGER REFERENCES iam_projects (db_id) ON DELETE CASCADE DEFERRABLE;
+UPDATE
+    iam_project_rules t
+SET
+    project_id = (
+        SELECT
+            db_id
+        FROM
+            iam_projects
+        WHERE
+            id = t.project_temp_id);
+ALTER TABLE iam_project_rules
+    DROP COLUMN project_temp_id;
+
+ALTER TABLE iam_staged_project_rules RENAME COLUMN project_id TO project_temp_id;
+ALTER TABLE iam_staged_project_rules
+    ADD COLUMN project_id INTEGER REFERENCES iam_projects (db_id) ON DELETE CASCADE DEFERRABLE;
+UPDATE
+    iam_staged_project_rules t
+SET
+    project_id = (
+        SELECT
+            db_id
+        FROM
+            iam_projects
+        WHERE
+            id = t.project_temp_id);
+ALTER TABLE iam_staged_project_rules
+    DROP COLUMN project_temp_id;
+
+-- fix functions
+CREATE OR REPLACE FUNCTION
+  query_rules_for_project(_project_id TEXT, _project_filter TEXT[])
+  RETURNS setof json AS $$
+
+  -- fetch staged rules
+   SELECT json_build_object(
+    'id', r.id,
+    'project_id', p.id,
```

Comment:
ℹ️ What this returns is the same, but it now needs to join the other table to get the text ID of the project.